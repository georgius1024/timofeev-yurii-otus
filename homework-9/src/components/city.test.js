import React from 'react'
import jest from 'jest-mock'
import renderer from 'react-test-renderer'
import City from './city'
const city = {
  country: 'RU',
  name: 'Moscow',
  lat: '55.75222',
  lng: '37.61556',
  image: '/images/11.jpg',
  temperature: '-7',
  humidity: '78',
  rain: '0',
  snow: '10',
  cloudiness: '80',
}
describe('City', () => {
  it('renders correctly', () => {
    const onRemove = jest.fn()
    const DOM = renderer.create(<City city={city} onRemove={onRemove} />)
    const html = JSON.stringify(DOM, null, 2)
    expect(html).toContain(city.name)
    expect(html).toContain(city.image)
    expect(html).toContain(city.lat)
    expect(html).toContain(city.lng)
    expect(DOM).toMatchSnapshot()
  })
})
