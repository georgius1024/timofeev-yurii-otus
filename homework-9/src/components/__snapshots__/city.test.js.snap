// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`City renders correctly 1`] = `
<div
  className="city"
>
  <div
    className="city-inner"
    style={
      Object {
        "backgroundImage": "url(/images/11.jpg)",
      }
    }
  >
    <div
      className="city-canvas"
    >
      <div
        style={
          Object {
            "textAlign": "right",
          }
        }
      >
        <a
          className="is-light has-text-white"
          href="https://maps.google.com/?q=55.75222,37.61556"
          target="_blank"
        >
          <span
            className="icon is-small  mr4"
          >
            <i
              className="fas fa-map"
            />
          </span>
          Open map
        </a>
      </div>
      <div
        className="header"
      >
        Moscow
      </div>
      <div
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        
      </div>
      <div
        className="column"
      >
        <div>
          Temperature: 
          N/A
          °C, feels like
           
          N/A
          °C
        </div>
        <div>
          Himidity: 
          %
        </div>
        <div>
          Pressure: 
          N/A
          mmHg
        </div>
        <div>
          Wind: 
          N/A
          m/s 
          N/A
        </div>
        <div>
          Local time: 
          N/A
        </div>
      </div>
    </div>
    <div
      className="removeControl"
    >
      <a
        className="button is-small is-danger is-outlined"
        onClick={[Function]}
      >
        Delete city
      </a>
    </div>
  </div>
</div>
`;
